src/ValueParsers/MonthNameProvider.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+namespace ValueParsers;
+
+/**
+ * A common interface for both localizing month names in formatters as well as unlocalizing month
+ * names in parsers.
+ *
+ * @since 0.8.3
+ *
+ * @licence GNU GPL v2+
+ * @author Thiemo Mättig
+ */
+interface MonthNameProvider {
+
+	/**
+	 * @param string $languageCode
+	 *
+	 * @return string[] Array mapping month numbers (1 to 12) to localized month names.
+	 */
+	public function getLocalizedMonthNames( $languageCode );
+
+	/**
+	 * @param string $languageCode
+	 *
+	 * @return int[] Array mapping localized month names (possibly including full month names,
+	 * genitive names and abbreviations) to month numbers (1 to 12).
+	 */
+	public function getMonthNumbers( $languageCode );
+
+}
```

Comment:
It seems to me like it would make more sense to have something like the following:

```
MonthNameProvider:
  getLocalizedMonthName( number ): string;
  getMonthNumber( string ): number | null;
```

I don't think users of this service will want to use different languages, so the language can be configured on construction. Also, they probably don't need to work with a lot of different months, probably just one.